# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/clang/gcc-path.patch
# Copyright (C) 2021 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---


--- clang-13.0.0.src/lib/Driver/ToolChains/Gnu.cpp.orig	2021-09-24 18:18:10.000000000 +0200
+++ clang-13.0.0.src/lib/Driver/ToolChains/Gnu.cpp	2021-10-01 17:11:15.769342526 +0200
@@ -2078,20 +2078,16 @@
   static const char *const AArch64Triples[] = {
-      "aarch64-none-linux-gnu", "aarch64-linux-gnu", "aarch64-redhat-linux",
-      "aarch64-suse-linux", "aarch64-linux-android"};
+      "aarch64-t2-linux-gnu", "aarch64-t2-linux-musl"};
   static const char *const AArch64beLibDirs[] = {"/lib"};
-  static const char *const AArch64beTriples[] = {"aarch64_be-none-linux-gnu",
-                                                 "aarch64_be-linux-gnu"};
+  static const char *const AArch64beTriples[] = {"aarch64_be-t2-linux-gnu",
+                                                 "aarch64_be-t2-linux-musl"};
 
   static const char *const ARMLibDirs[] = {"/lib"};
-  static const char *const ARMTriples[] = {"arm-linux-gnueabi",
-                                           "arm-linux-androideabi"};
-  static const char *const ARMHFTriples[] = {"arm-linux-gnueabihf",
-                                             "armv7hl-redhat-linux-gnueabi",
-                                             "armv6hl-suse-linux-gnueabi",
-                                             "armv7hl-suse-linux-gnueabi"};
+  static const char *const ARMTriples[] = {"arm-t2-linux-gnueabi",
+                                           "arm-t2-linux-musl"};
+  static const char *const ARMHFTriples[] = {"arm-t2-linux-gnueabihf"};
   static const char *const ARMebLibDirs[] = {"/lib"};
-  static const char *const ARMebTriples[] = {"armeb-linux-gnueabi",
-                                             "armeb-linux-androideabi"};
+  static const char *const ARMebTriples[] = {"armeb-t2-linux-gnueabi",
+                                             "armeb-t2-linux-gnu"};
   static const char *const ARMebHFTriples[] = {
-      "armeb-linux-gnueabihf", "armebv7hl-redhat-linux-gnueabi"};
+      "armeb-t2-linux-gnueabihf"};
 
@@ -2102,8 +2098,3 @@
   static const char *const X86_64Triples[] = {
-      "x86_64-linux-gnu",       "x86_64-unknown-linux-gnu",
-      "x86_64-pc-linux-gnu",    "x86_64-redhat-linux6E",
-      "x86_64-redhat-linux",    "x86_64-suse-linux",
-      "x86_64-manbo-linux-gnu", "x86_64-linux-gnu",
-      "x86_64-slackware-linux", "x86_64-unknown-linux",
-      "x86_64-amazon-linux",    "x86_64-linux-android"};
+      "x86_64-t2-linux-gnu",    "x86_64-t2-linux-musl"};
   static const char *const X32Triples[] = {"x86_64-linux-gnux32",
@@ -2113,7 +2104,6 @@
   static const char *const X86Triples[] = {
-      "i586-linux-gnu",     "i686-linux-gnu",
-      "i686-pc-linux-gnu",  "i386-redhat-linux6E",
-      "i686-redhat-linux",  "i386-redhat-linux",
-      "i586-suse-linux",    "i686-montavista-linux",
-      "i686-linux-android", "i686-gnu",
+      "i386-t2-linux-gnu",  "i486-t2-linux-gnu",  "i586-t2-linux-gnu",
+      "i686-t2-linux-gnu",
+      "i386-t2-linux-musl", "i486-t2-linux-musl", "i586-t2-linux-musl",
+      "i686-t2-linux-musl"
   };
@@ -2126,8 +2116,6 @@
   static const char *const MIPSTriples[] = {
-      "mips-linux-gnu", "mips-mti-linux", "mips-mti-linux-gnu",
-      "mips-img-linux-gnu", "mipsisa32r6-linux-gnu"};
+      "mips-t2-linux-gnu", "mips-t2-linux-musl"};
   static const char *const MIPSELLibDirs[] = {"/lib"};
   static const char *const MIPSELTriples[] = {
-      "mipsel-linux-gnu", "mips-img-linux-gnu", "mipsisa32r6el-linux-gnu",
-      "mipsel-linux-android"};
+      "mipsel-t2-linux-gnu", "mipsel-t2-linux-musl"};
 
@@ -2135,18 +2123,13 @@
   static const char *const MIPS64Triples[] = {
-      "mips64-linux-gnu",      "mips-mti-linux-gnu",
-      "mips-img-linux-gnu",    "mips64-linux-gnuabi64",
-      "mipsisa64r6-linux-gnu", "mipsisa64r6-linux-gnuabi64"};
+      "mips64-t2-linux-gnu",   "mips64-t2-linux-musl"};
   static const char *const MIPS64ELLibDirs[] = {"/lib64", "/lib"};
   static const char *const MIPS64ELTriples[] = {
-      "mips64el-linux-gnu",      "mips-mti-linux-gnu",
-      "mips-img-linux-gnu",      "mips64el-linux-gnuabi64",
-      "mipsisa64r6el-linux-gnu", "mipsisa64r6el-linux-gnuabi64",
-      "mips64el-linux-android"};
+      "mips64el-t2-linux-gnu",   "mips64el-t2-linux-musl"};
 
   static const char *const MIPSN32LibDirs[] = {"/lib32"};
-  static const char *const MIPSN32Triples[] = {"mips64-linux-gnuabin32",
-                                               "mipsisa64r6-linux-gnuabin32"};
+  static const char *const MIPSN32Triples[] = {"mips64-t2-linux-gnuabin32",
+                                               "mips64-t2-linux-muslabin32"};
   static const char *const MIPSN32ELLibDirs[] = {"/lib32"};
   static const char *const MIPSN32ELTriples[] = {
-      "mips64el-linux-gnuabin32", "mipsisa64r6el-linux-gnuabin32"};
+      "mips64el-t2-linux-gnuabin32", "mips64el-t2-linux-muslabin32"};
 
@@ -2157,10 +2140,6 @@
   static const char *const PPCTriples[] = {
-      "powerpc-linux-gnu", "powerpc-unknown-linux-gnu", "powerpc-linux-gnuspe",
-      // On 32-bit PowerPC systems running SUSE Linux, gcc is configured as a
-      // 64-bit compiler which defaults to "-m32", hence "powerpc64-suse-linux".
-      "powerpc64-suse-linux", "powerpc-montavista-linuxspe"};
+      "powerpc-t2-linux-gnu", "powerpc-t2-linux-musl"};
   static const char *const PPCLELibDirs[] = {"/lib32", "/lib"};
-  static const char *const PPCLETriples[] = {"powerpcle-linux-gnu",
-                                             "powerpcle-unknown-linux-gnu",
-                                             "powerpcle-linux-musl"};
+  static const char *const PPCLETriples[] = {"powerpcle-t2-linux-gnu",
+                                             "powerpcle-t2-linux-musl"};
 
@@ -2168,27 +2147,20 @@
   static const char *const PPC64Triples[] = {
-      "powerpc64-linux-gnu", "powerpc64-unknown-linux-gnu",
-      "powerpc64-suse-linux", "ppc64-redhat-linux"};
+      "powerpc64-t2-linux-gnu", "powerpc64-t2-linux-msul"};
   static const char *const PPC64LELibDirs[] = {"/lib64", "/lib"};
   static const char *const PPC64LETriples[] = {
-      "powerpc64le-linux-gnu", "powerpc64le-unknown-linux-gnu",
-      "powerpc64le-none-linux-gnu", "powerpc64le-suse-linux",
-      "ppc64le-redhat-linux"};
+      "powerpc64le-t2-linux-gnu", "powerpc64le-t2-linux-musl"};
 
   static const char *const RISCV32LibDirs[] = {"/lib32", "/lib"};
-  static const char *const RISCV32Triples[] = {"riscv32-unknown-linux-gnu",
-                                               "riscv32-linux-gnu",
-                                               "riscv32-unknown-elf"};
+  static const char *const RISCV32Triples[] = {"riscv32-t2-linux-gnu",
+                                               "riscv32-t2-linux-musl"};
   static const char *const RISCV64LibDirs[] = {"/lib64", "/lib"};
-  static const char *const RISCV64Triples[] = {"riscv64-unknown-linux-gnu",
-                                               "riscv64-linux-gnu",
-                                               "riscv64-unknown-elf",
-                                               "riscv64-redhat-linux",
-                                               "riscv64-suse-linux"};
+  static const char *const RISCV64Triples[] = {"riscv64-t2-linux-gnu",
+                                               "riscv64-t2-linux-musl"};
 
   static const char *const SPARCv8LibDirs[] = {"/lib32", "/lib"};
-  static const char *const SPARCv8Triples[] = {"sparc-linux-gnu",
-                                               "sparcv8-linux-gnu"};
+  static const char *const SPARCv8Triples[] = {"sparc-t2-linux-gnu",
+                                               "sparc-t2-linux-musl"};
   static const char *const SPARCv9LibDirs[] = {"/lib64", "/lib"};
-  static const char *const SPARCv9Triples[] = {"sparc64-linux-gnu",
-                                               "sparcv9-linux-gnu"};
+  static const char *const SPARCv9Triples[] = {"sparc64-t2-linux-gnu",
+                                               "sparc64-t2-linux-musl"};
 
@@ -2196,4 +2168,3 @@
   static const char *const SystemZTriples[] = {
-      "s390x-linux-gnu", "s390x-unknown-linux-gnu", "s390x-ibm-linux-gnu",
-      "s390x-suse-linux", "s390x-redhat-linux"};
+      "s390x-t2-linux-gnu", "s390x-t2-linux-musl"};
 
